### PR TITLE
Michael/ir 2479 trigger

### PR DESCRIPTION
**Pull request by MbfloydIR**
## Summary
Added logic to check/wait for the collider component's collider existence before trying to make it a trigger

## References
closes #_insert number here_

### CODE REVIEW

**Comment by HexaField** [comment]
@cla-bot check

File: packages/spatial/src/physics/components/TriggerComponent.ts [hunk 1]
```diff
@@ -91,14 +91,14 @@ export const TriggerComponent = defineComponent({
     const colliderComponent = useComponent(entity, ColliderComponent)
 
     useEffect(() => {
-      if (!colliderComponent?.value) return
+      if (!colliderComponent?.value || !colliderComponent?.hasCollider.value) return
```
Comment (HexaField) [inline packages/spatial/src/physics/components/TriggerComponent.ts]: we should just move this useEffect to the collider component

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/spatial/src/physics/components/TriggerComponent.ts (hunk 1 above)
Comment (MbfloydIR) [reply packages/spatial/src/physics/components/TriggerComponent.ts]: implemented

File: packages/spatial/src/physics/components/ColliderComponent.tsx [hunk 2]
```diff
@@ -49,7 +50,8 @@ export const ColliderComponent = defineComponent({
       friction: 0.5,
       restitution: 0.5,
       collisionLayer: CollisionGroups.Default,
-      collisionMask: DefaultCollisionMask
+      collisionMask: DefaultCollisionMask,
+      hasCollider: false
```
Comment (HexaField) [inline packages/spatial/src/physics/components/ColliderComponent.tsx]: let's scope hasCollider to the reactor as to not expose it on the component properties

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/spatial/src/physics/components/ColliderComponent.tsx (hunk 2 above)
Comment (MbfloydIR) [reply packages/spatial/src/physics/components/ColliderComponent.tsx]: implemented

**Review by HexaField — APPROVED** [review]

**Comment by HexaField** [comment]
Just need to fix tests. Likely need to ensure reactors have run, or change the order of setComponents to have the trigger before the collider

**Pull request merged**